### PR TITLE
Add V2 funding cycle metadata decoder

### PR DESCRIPTION
**Pull request by aeolianeth**
## What does this PR do and why?

This PR does 3 things:

- adds a metadata decoder for V2 metadata (with an accompanying test)
- renames the unit formatter functions to be slightly more grok-able (and adds bulk explanatory JSDoc)
- fixes unit conversion in V2 metadata serlializer.

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the
changes._

## Acceptance checklist

- [x] I have evaluated the [Approval Guidelines](../CONTRIBUTING.md#approval-guidelines) for this PR.
- [x] I have tested this PR in [all supported browsers](../CONTRIBUTING.md#browser-support).

### CODE REVIEW

**Review by johnnyd-eth — APPROVED** [review]
Sickkk @aeolianeth this all looks great. Haven't tested it but will do it on my pay button branch once we merge this

File: src/utils/formatNumber.ts [hunk 1]
```diff
@@ -73,6 +156,11 @@ const separateThousands = (str?: string, separator = thousandsSeparator) => {
   return str.replace(/\B(?=(\d{3})+(?!\d))/g, separator)
 }
 
+// Strips string of all commas
```
Comment (aeolianeth) [inline src/utils/formatNumber.ts:159]: Just moved this function from above to here

File: src/utils/v2/serializers.ts [hunk 2]
```diff
@@ -41,16 +45,17 @@ export type SerializedV2FundAccessConstraint = Record<
 export const serializeV2FundingCycleMetadata = (
   fundingCycleMetadata: V2FundingCycleMetadata,
 ): SerializedV2FundingCycleMetadata => ({
-  reservedRate: fromPermille(fundingCycleMetadata.reservedRate),
-  redemptionRate: fromPerbicent(fundingCycleMetadata.redemptionRate),
-  ballotRedemptionRate: fromPerbicent(
+  reservedRate: permyriadToPercent(fundingCycleMetadata.reservedRate),
```
Comment (aeolianeth) [inline src/utils/v2/serializers.ts:48]: these are fixes

File: src/utils/v2/serializers.ts [hunk 3]
```diff
@@ -63,17 +68,20 @@ export const serializeV2FundingCycleMetadata = (
 
 export const deserializeV2FundingCycleMetadata = (
   serializedFundingCycleMetadata: SerializedV2FundingCycleMetadata,
-): V2FundingCycleMetadata => ({
-  reservedRate: parsePermille(serializedFundingCycleMetadata.reservedRate),
-  redemptionRate: parsePerbicent(serializedFundingCycleMetadata.redemptionRate),
-  ballotRedemptionRate: parsePerbicent(
+): Omit<V2FundingCycleMetadata, 'version'> => ({
+  reservedRate: percentToPermyriad(serializedFundingCycleMetadata.reservedRate),
```
Comment (aeolianeth) [inline src/utils/v2/serializers.ts:72]: related fixes

**Comment by aeolianeth** [comment]
> Sickkk @aeolianeth this all looks great. Haven't tested it but will do it on my pay button branch once we merge this

Thanks @johnnyd-eth ! I made a few changes since you last looked at this.

@peripheralist could you please give the final review? This PR appears big but it's mostly the added test, JSDoc and renaming changes.

File: src/models/v2/fundingCycle.ts [hunk 4]
```diff
@@ -16,6 +17,7 @@ export type V2FundingCycleMetadata = {
   pauseRedeem: boolean
   pauseMint: boolean
   pauseBurn: boolean
+  allowChangeToken: boolean
```
Comment (aeolianeth) [inline src/models/v2/fundingCycle.ts:20]: Adding this missed field

**Comment by aeolianeth** [comment]
@peripheralist 👋 friendly ping here, could you pls review this?

**Comment by aeolianeth** [comment]
Gonna merge this, only impacts to V1 is renaming functions which is low-risk.

cc @peripheralist @johnnyd-eth in case we need to revert.

**Pull request merged**